app/components/Views/Notifications/Details/index.tsx

```diff
@@ -0,0 +1,138 @@
+import Header from './Title';
+import { createStyles } from './styles';
+import ModalField from './Fields';
+import ModalHeader from './Headers';
+import ModalFooter from './Footers';
+
+interface Props {
+  navigation: NavigationProp<ParamListBase>;
+  route: {
+    params: {
+      notification: Notification;
+    };
+  };
+}
+
+const NotificationsDetails = ({ route }: Props) => {
+  const { notification } = route.params;
+  const { markNotificationAsRead } = useMarkNotificationAsRead();
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(true);
+
+  const theme = useTheme();
+  const styles = createStyles(theme);
+
+  // Effect - Mark As Read, on open
+  useEffect(() => {
+    if (!notification.isRead) {
+      markNotificationAsRead([
+        {
+          id: notification.id,
+          type: notification.type,
+          isRead: notification.isRead,
+        },
+      ]);
+    }
+  }, [notification, markNotificationAsRead]);
+
+  const state =
+    NotificationComponentState[notification?.type]?.createModalDetails?.(
+      notification,
+    );
+
+  if (!state) {
+    return null;
+  }
+
+  return (
+    <ScrollView contentContainerStyle={styles.contentContainerWrapper}>
+      <View style={styles.renderContainer}>
+        {/* Modal Headers */}
+        {state.header && <ModalHeader modalHeader={state.header} />}
+
+        {/* Modal Fields */}
+        {state.fields.map((field, idx) => (
+          <ModalField
+            key={idx}
+            modalField={field}
+            isCollapsed={isCollapsed}
+            setIsCollapsed={setIsCollapsed}
+          />
+        ))}
+
+        {/* Modal Footers */}
+        {state.footer && <ModalFooter modalFooter={state.footer} />}
+      </View>
+    </ScrollView>
+  );
+};
+
+export default NotificationsDetails;
+
+NotificationsDetails.navigationOptions = ({
+  route,
+  navigation,
+}: {
+  route: RouteProp<{ params: { notification: Notification } }, 'params'>;
+  navigation: NavigationProp<Record<string, undefined>>;
+}) => {
+  const notification = route?.params?.notification;
+  if (!notification) {
+    navigation.goBack();
+    return {};
+  }
+
+  const state =
+    NotificationComponentState[notification?.type]?.createModalDetails?.(
+      notification,
+    );
+  if (!state) {
+    navigation.goBack();
+    return {};
+  }
+
+  const styles = StyleSheet.create({
+    backIcon: {
+      marginHorizontal: 16,
+    },
+  });
+
+  return {
+    // eslint-disable-next-line react/display-name
+    headerLeft: () => (
+      <TouchableOpacity onPress={() => navigation.goBack()}>
+        <Icon
+          name={IconName.ArrowLeft}
+          size={IconSize.Md}
+          style={styles.backIcon}
+        />
+      </TouchableOpacity>
+    ),
+    // eslint-disable-next-line react/display-name
+    headerTitle: () => (
+      <Header
+        title={state.title}
+        subtitle={formatIsoDateString(state.createdAt)}
+      />
+    ),
+  };
```

Comment:
We could remove this eslint-disable-next-line by creating components for headerLeft and header right I believe, something like this:

I also believe the headerLeft could be memoized because it shouldn't change notification to notification, I will defer the decision of the headerTitle to you, since you have more context on if it changes a lot or not



```suggestion
const HeaderLeft = useCallback(() => (
    <TouchableOpacity onPress={() => navigation.goBack()}>
      <Icon
        name={IconName.ArrowLeft}
        size={IconSize.Md}
        style={styles.backIcon}
      />
    </TouchableOpacity>
  ), [navigation]);

// Only use useCallback here if the state.title is most likely always the same, if not, it is unnecessary to memoize it
  const HeaderTitle = useCallback(() => (
    <Header
      title={state.title}
      subtitle={formatIsoDateString(state.createdAt)}
    />
  ), [state.title, state.createdAt]);
  
    return {
    headerLeft: HeaderLeft,
    headerTitle: HeaderTitle,
  };
```